sound/soc/sof/ipc.c

```diff
@@ -249,15 +249,15 @@ static int tx_wait_done(struct snd_sof_ipc *ipc, struct snd_sof_ipc_msg *msg,
 				 msecs_to_jiffies(sdev->ipc_timeout));
 
 	if (ret == 0) {
-		dev_err(sdev->dev, "error: ipc timed out for 0x%x size %d\n",
+		dev_err(sdev->dev, "ipc tx timed out for 0x%x size %d\n",
 			hdr->cmd, hdr->size);
 		snd_sof_handle_fw_exception(ipc->sdev);
 		ret = -ETIMEDOUT;
 	} else {
 		ret = msg->reply_error;
 		if (ret < 0) {
-			dev_err(sdev->dev, "error: ipc error for 0x%x size %zu\n",
-				hdr->cmd, msg->reply_size);
+			dev_err(sdev->dev, "ipc tx error for 0x%x: %d\n",
+				hdr->cmd, ret);
```

Comment:
why not put both error code and error size? doesn't cost us anything, and might help us detect invalid configuration, either bad code or bad size.

Reply:
Right. I have found this print odd to be honest:
`error: ipc error for 0x500154cd size 384`
the size is not the size of the message, but the expected or provided RX buffer size. In case of timeout the `size` is the size of the message and not the size of the reply (expected/provided).

We print error in the reply handling if the reply size had mismatch (after PR #3108 if there is not enough space provided), I don't see the caller provided size to be useful here that much. But I might be completely wrong on this.

We should move to '%#x' here as well.

Reply:
I have also zero understanding of errors/sizes, so by default I'd add all the information and ask firmware folks to look into this...